box-ui/src/components/ts/SelectButton.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { FunctionComponent } from 'react';
+import { FormControl, InputLabel, Select, MenuItem } from '@mui/material';
+import { SelectButtonProps } from '../../utils/Props';
+
+const SelectButton: FunctionComponent<SelectButtonProps> = (props: SelectButtonProps) => {
+    return (
+        <div {...props.div} className='SelectButton'>
+            <FormControl variant='filled' sx={{ m: 1, minWidth: 140 }}>
+                <InputLabel id='input-label' sx={props.selectItems.inputLabel.style}>
+                    {props.selectItems.inputLabel.text}
+                </InputLabel>
+                <Select
+                    sx={props.menuItemsStyle}
+                    defaultValue=''
+                    onChange={(e) => (props.onChange ? props.onChange(e) : null)}
+                    onClick={(e) => (props.onClick ? props.onClick(e) : null)}
+                >
+                    {props.selectItems.menuItems.map((menuItemText, index) => (
+                        <MenuItem key={index} value={menuItemText ? menuItemText : ''}>
```

Comment:
Using the index as a key is an anti-pattern you could add an id argument to each menuItem. https://medium.com/wesionary-team/using-index-as-a-key-is-an-anti-pattern-in-react-8e5db3aea3a9